src/main/java/seedu/command/AddCommand.java

```diff
@@ -1,36 +1,110 @@
 package seedu.command;
 
-import com.google.gson.Gson;
+import seedu.module.Lesson;
+import seedu.online.NusMods;
+import seedu.storage.ModStorage;
 import seedu.module.Module;
-import seedu.timetable.TimeTable;
+import seedu.module.Semester;
+import seedu.ui.TextUi;
+import java.util.ArrayList;
 
-import java.io.Reader;
-import java.nio.file.Files;
-import java.nio.file.Paths;
+import java.io.IOException;
 
 public class AddCommand extends Command {
-
+    private final Integer BALANCE_ARRAY = 1;
     private final String moduleCode;
-    //public TimeTable timetable = new TimeTable();
+    private final Integer semester;
 
-    public AddCommand(String moduleCode) {
+    public AddCommand(String moduleCode, Integer semester) {
         this.moduleCode = moduleCode;
+        this.semester = semester;
     }
 
     public void execute() {
-        /*
-        String path = "data/Modules/" + moduleCode + ".json";
+        TextUi.printAddMessage(moduleCode);
+        Module module = new Module(moduleCode);
+        try {
+            module = NusMods.fetchModOnline(moduleCode);
+            System.out.println("fetchOnlineWorks");
+        } catch (IOException e1) {
+            TextUi.printNoConnectionMessage();
+            try {
+                module = ModStorage.loadModInfo(moduleCode);
+                System.out.println("fetchOfflineWorks");
+            } catch (IOException e2) {
+                TextUi.printNotFoundMessage();
+            }
+        }
+
+        ArrayList<Lesson> addedLesson = new ArrayList<>();
+        Semester semesterData = module.getSemester(semester);
+        System.out.println(semesterData.isTutorialExist());
+        if (semesterData.isLectureExist()) {
+            Lesson lecture = selectLecture(semesterData);
+            addedLesson.add(lecture);
+        }
+        if (semesterData.isTutorialExist()) {
+            Lesson tutorial = selectTutorial(semesterData);
+            addedLesson.add(tutorial);
+        }
+        if (semesterData.isLabExist()) {
+            Lesson lab = selectLab(semesterData);
+            addedLesson.add(lab);
+        }
+        Module addedModule = module;
+        addedModule.getSemester(semester).changeTimetable(addedLesson);
```

Comment:
Would this work  with modules that have multiple lectures or multiple tutorials like CS2113T?